src/libraries/System.Private.CoreLib/src/System/Utf8String.Construction.cs

```diff
@@ -341,7 +341,7 @@ internal static Utf8String CreateFromRune(Rune value)
             // "skip validation" transcoder because the caller could've mutated the input buffer between the
             // initial counting step and the transcoding step below.
 
-            status = Utf8.FromUtf16(value, newBuffer, out _, out int bytesWritten, replaceInvalidSequences: false);
+            status = Utf8.FromUtf16(value, newBuffer.AsSpan(0, newBuffer.Length - 1), out _, out int bytesWritten, replaceInvalidSequences: false);
             if (status != OperationStatus.Done || bytesWritten != newBuffer.Length)
```

Comment:
Does this need to be `bytesWritten != newBuffer.Length - 1`?

Reply:
Yep - good catch. I added a test for this as well, as it looks like it wasn't covered before.